### PR TITLE
Fix scanner for Windows line endings and add tests

### PR DESCRIPTION
**Pull request by jen20**
I think this should be enough to fix hashicorp/terraform#4069, though this is not yet verified and changes to the parser may also be necessary.

### CODE REVIEW

File: hcl/scanner/scanner.go [hunk 1]
```diff
@@ -224,7 +224,11 @@ func (s *Scanner) scanComment(ch rune) {
 	// single line comments
 	if ch == '#' || (ch == '/' && s.peek() != '*') {
 		ch = s.next()
-		for ch != '\n' && ch >= 0 && ch != eof {
+		for ch != '\n' && ch != '\r' && ch >= 0 && ch != eof {
+			if ch == '\r' && s.peek() == '\n' {
```
Comment (phinze) [inline hcl/scanner/scanner.go]: Confused about how if the guard has `&& ch != '\r'` this conditional will ever be true.

Comment (phinze) [inline hcl/scanner/scanner.go]: Oh sorry - by "the guard" I mean the one on L227.

Comment (phinze) [inline hcl/scanner/scanner.go]: In other words, I believe this will break comments immediately on `\r` instead of only on `\r\n`... which might be fine.

Comment (jen20) [inline hcl/scanner/scanner.go]: That's a typo (originally the whole guard was one line). Will fix.

Comment (phinze) [inline hcl/scanner/scanner.go]: :+1:

Comment (jen20) [inline hcl/scanner/scanner.go]: The question is do we want the `\r` to be included in the comment or treated as part of the line break? Unlike HEREDOCs I'd say they should be part of the line break for comments (though unless we're going to process comments it doesn't make much difference).

**Comment by mitchellh** [comment]
On first pass something about this looks a little weird. 

The `ch != 'r'` test doesn't seem necessary in the for loop.

I didn't check the rest of the code but I believe due to how the scanner works the newline character(s) still show up in the Token text correct? I'm not sure if this is okay or not but it is worth questioning to make sure it is correct.

**Comment by jen20** [comment]
@mitchellh The loop was a typo - now fixed. The current situation with this PR is:
- comments have the `\r` as part of the token text
- HEREDOCs keep the `\r` throughout (which is probably OK - unless we want to handle line ending conversion in e.g. Windows provisioners), but the `\r` can be part of the anchor so they terminate correctly.

**Comment by phinze** [comment]
@jen20 and I both reviewed pre-rewrite behavior and confirmed that the behavior as he just described was is the same. this LGTM

**Pull request merged**